### ISSUE TITLE
refmt: Stack overflow when using scoped record deconstrutor

### ISSUE THREAD

**Issue by jez** [facebook/reason#2400]
I'm seeing a `Stack overflow` when trying to convert an OCaml file to a Reason file.

It's possible I'm using a feature of OCaml that `refmt` doesn't support?

### Steps to reproduce

Consider this file:

```ocaml
(* foo.ml *)
module A = struct
  type t = {
    x: string;
  }
end

let x =
  let A.{ x } = {x = "foo"} in
  print_endline x
```

It is valid OCaml:

```bash
❯ ocaml foo.ml
foo
```

It does not convert successfully:

```bash
❯ refmt -p re foo.ml
refmt: internal error, uncaught exception:
       Stack overflow
```

### Versions

```bash
❯ ocaml --version
The OCaml toplevel, version 4.07.1

❯ refmt --version
Reason 3.3.7 @ 43efc14
```

**Comment by jez** [comment]
After spending all of 30 seconds evaluating how Reason looks on my project I think I'm going to stick with OCaml syntax, so no urgency on fixing this. Just thought it might be good to track. Feel free to close if it's not useful.

**Comment by anmonteiro** [comment]
Duplicate of https://github.com/facebook/reason/issues/2301, fixed in #2314
